src/api/objects/ObjectAPI.js

```diff
@@ -0,0 +1,232 @@
+     * @param {module:openmct.DomainObject} domainObject the domain object to
+     *        delete
+     * @returns {Promise} a promise which will resolve when the domain object
+     *          has been deleted, or be rejected if it cannot be deleted
+     */
+
+    /**
+     * Get a domain object.
+     *
+     * @method get
+     * @memberof module:openmct.ObjectProvider#
+     * @param {string} key the key for the domain object to load
+     * @returns {Promise} a promise which will resolve when the domain object
+     *          has been saved, or be rejected if it cannot be saved
+     */
+
+    [
+        'save',
+        'delete',
+        'get'
+    ].forEach(function (method) {
+        ObjectAPI.prototype[method] = function () {
+            var key = arguments[0],
+                provider = this.getProvider(key);
+
+            if (!provider) {
+                throw new Error('No Provider Matched');
+            }
+
+            if (!provider[method]) {
+                throw new Error('Provider does not support [' + method + '].');
+            }
+
+            return provider[method].apply(provider, arguments);
+        };
+    });
+
+    /**
+     * Add a root-level object.
+     * @param {module:openmct.DomainObject} domainObject the root-level object
+     *        to add.
+     * @method addRoot
+     * @memberof module:openmct.ObjectAPI#
+     */
+    ObjectAPI.prototype.addRoot = function (key) {
+        this.rootRegistry.unshift(key);
+    };
+
+    /**
+     * Remove a root-level object.
+     * @param {module:openmct.ObjectAPI~Identifier} id the identifier of the
+     *        root-level object to remove.
+     * @method removeRoot
+     * @memberof module:openmct.ObjectAPI#
+     */
+    ObjectAPI.prototype.removeRoot = function (key) {
+        this.rootRegistry = this.rootRegistry.filter(function (k) {
+            return (
+                k.identifier !== key.identifier ||
+                k.namespace !== key.namespace
+            );
+        });
+    };
+
+    /**
+     * Modify a domain object.
+     * @param {module:openmct.DomainObject} object the object to mutate
+     * @param {string} path the property to modify
+     * @param {*} value the new value for this property
+     * @method mutate
+     * @memberof module:openmct.ObjectAPI#
+     */
+    ObjectAPI.prototype.mutate = function (domainObject, path, value) {
+        return new MutableObject(domainObject).set(path, value);
+    };
+
+    /**
+     * Observe changes to a domain object.
+     * @param {module:openmct.DomainObject} object the object to observe
+     * @param {string} path the property to observe
+     * @param {Function} callback a callback to invoke when new values for
+     *        this property are observed
+     * @method observe
+     * @memberof module:openmct.ObjectAPI#
+     */
+    ObjectAPI.prototype.observe = function (domainObject, path, callback) {
+        return new MutableObject(domainObject).on(path, callback);
```

Comment:
This is going to lead to memory leaks in the current implementation as `on` doesn't return an unlistener. The intention was for `.stopListening()` to be called, which will unregister all listeners at once. This isn't possible in this implementation because the `MutableObject` instance isn't returned. A simple solution for now is to return an unlistener from `MutableObject.on()`